From fae0caf73ff5e582e39ed8173b407d8a6a5cde3d Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 22 Jul 2015 15:20:09 +0100
Subject: [PATCH 2/4] Fix pool_allocator compile-guards

With MinGW-w64 the template<class Other> variant was not
defined.
---
 hlslang/Include/PoolAlloc.h | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/hlslang/Include/PoolAlloc.h b/hlslang/Include/PoolAlloc.h
index 3418c88..3312221 100644
--- a/hlslang/Include/PoolAlloc.h
+++ b/hlslang/Include/PoolAlloc.h
@@ -140,18 +140,15 @@ public:
    }
 #endif
 
-#if defined(_MSC_VER) && _MSC_VER >= 1300
    template<class Other>
+#if defined(_MSC_VER) && _MSC_VER >= 1300
 #ifdef USING_SGI_STL
       pool_allocator(const pool_allocator<Other>& p) /*: allocator(p.getAllocator())*/ { }
 #else
       pool_allocator(const pool_allocator<Other>& p) : allocator(&p.getAllocator()) { }
 #endif
-#endif
-
-#ifndef _WIN32
-      template<class Other>
-         pool_allocator(const pool_allocator<Other>& p) : allocator(&p.getAllocator()) { }
+#else
+      pool_allocator(const pool_allocator<Other>& p) : allocator(&p.getAllocator()) { }
 #endif
 
 #ifdef USING_SGI_STL
-- 
2.5.0

